fortimanager_connector.py

```diff
@@ -0,0 +1,462 @@
+
+        if response_code == 0:
+            self.save_progress("Test Connectivity Passed")
+            return action_result.set_status(phantom.APP_SUCCESS)
+        else:
+            self.save_progress("response_data: {}".format(response_data))
+            self.save_progress("Test Connectivity Failed.")
+            return action_result.set_status(phantom.APP_ERROR, "Test Connectivity Failed")
+
+    def _handle_create_firewall_policy(self, param):
+
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        fmg_instance = None
+        level = param['level']
+        pkg = param['package']
+        data = {
+                    'name': param['name'],
+                    'srcintf': param['source_interface'],
+                    'dstintf': param['destination_interface'],
+                    'service': param['service'],
+                    'srcaddr': param['source_address'],
+                    'dstaddr': param['destination_address'],
+                    'action': param['action'],
+                    'status': param['status'],
+                    'inspection-mode': param['inspection_mode'],
+                    'logtraffic': param['log_traffic'],
+                    'schedule': param['schedule']
+                }
+        if level == 'ADOM':
+            adom = param.get('adom')
+            if not adom:
+                adom = 'root'
+            endpoint = ADOM_FIREWALL_ENDPOINT.format(adom=adom, pkg=pkg)
+        elif level == 'Global':
+            adom = 'global'
+            policy_type = param['policy_type']
+            endpoint = GLOBAL_FIREWALL_ENDPOINT.format(pkg=pkg, policy_type=policy_type)
+
+        try:
+            fmg_instance = self._login(action_result)
+            fmg_instance.lock_adom(adom)
+            response_code, response_data = fmg_instance.add(endpoint, **data)
+            fmg_instance.commit_changes(adom)
+            fmg_instance.unlock_adom(adom)
+            fmg_instance.logout()
+        except Exception as e:
+            error_msg = self._get_error_msg_from_exception(e)
+            self.save_progress("Create Firewall Policy action failed")
+            self.debug_print("Create Firewall Policy action failed: {}".format(error_msg))
+            return action_result.set_status(phantom.APP_ERROR, None)
+
+        if response_code == 0:
+            action_result.add_data(response_data)
+            summary = action_result.update_summary({})
+            summary['status'] = 'Successfully added firewall policy'
+            return action_result.set_status(phantom.APP_SUCCESS)
+        else:
+            self.save_progress("Failed.")
+            error_msg = response_data['status']['message']
+            return action_result.set_status(phantom.APP_ERROR, "Failed to create firewall policy. Reason: {}".format(error_msg))
+
+    def _handle_list_firewall_policies(self, param):
```

Comment:
Just for clarity - are we leaving the Global options in the code but disabling them in the config options, or should we be removing global entirely?

Reply:
I think it might be best to remove them from the code as well as in the config

Reply:
I would like to just comment them out for ease of future development when we do add the feature. FYI this portion of the code is unreachable given the config options.